transport_nantes/topicblog/forms.py

```diff
@@ -5,7 +5,45 @@
 from mailing_list.models import MailingList
 
 
-class TopicBlogItemForm(ModelForm):
+class ModelFormTemplateList(ModelForm):
+    """Render a ModelForm and pre-select template_name field's initial value.
+
+    If a value was already selected (e.g. in case of edit), the value is left
+    alone and preselected.
+    In case of a new object, the 1st template is pre-selected.
+
+    This class implies an implementation of template_config and that the
+    instance has a template_name attribute.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.set_initial_template()
+
+    def get_template_list(self) -> list:
+
+        templates = self.instance.template_config
+        template_list = \
+            [(None, "Selectionnez un template ...")]
+        for template, value in templates.items():
+            template_list.append((template, value["user_template_name"]))
+
+        return template_list
+
+    def set_initial_template(self):
+        template_list = self.get_template_list()
+        if self.instance.template_name == "" and len(template_list) >= 2:
+            self.fields['template_name'] = forms.ChoiceField(
+                choices=template_list,
+                initial=template_list[1][0],
```

Comment:
Do I correctly understand that we are arbitrarily picking the first element of the list as the default?
* Shouldn't you filter by the `active` flag?
* Might it not be more appropriate to add a `default_choice: True` and use that (or the first instance of that) if it exists?
* Is it possible that thie template config has become complex enough that it should be a class (a second issue, second PR, no doubt)?

Reply:
All good suggestions ! The third one (making it class) is what we originally went for before deciding a dict was better. 
We can open an issue but because this isn't really blocking for now, I suggest we stick with that and work on more productive issues.

I'll rework the template_config so it uses active and default_choice

Reply:
IIRC, it was originally a model.  I'm proposing it be a class so that it can control its interface and help us avoid errors, have methods like `is_active()` instead of all the code having to know that we stored the data at just this one place in this increasingly complex dict.

Reply:
Agreed, it's not blocking for now.